### PR TITLE
Use pkg/errors instead of fmt.Errorf

### LINKED ISSUE #689: cleanup: use pkg/errors instead of fmt.Errorf
Until go 2.0 is out and/or until the code base transitions to the updates in error handling it has, a suggested change discussed in the recent Zoom meeting is to replace the usage of the construct `fmt.Errorf()` in favor of the wrappers available in the package [pkg/errors](https://godoc.org/github.com/pkg/errors)

a github search reveals:
https://github.com/kubernetes-sigs/cluster-api/search?q=%22fmt.Errorf%22&unscoped_q=%22fmt.Errorf%22

/kind cleanup
/priority backlog
/help

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #689

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: cmd/clusterctl/clusterdeployer/bootstrap/minikube/minikube.go [hunk 1]
```diff
@@ -61,7 +62,7 @@ var minikubeExec = func(env []string, args ...string) (string, error) {
 	cmdOut, err := cmd.CombinedOutput()
 	klog.V(2).Infof("Ran: %v %v Output: %v", executable, args, string(cmdOut))
 	if err != nil {
-		err = fmt.Errorf("error running command '%v %v': %v", executable, strings.Join(args, " "), err)
+		err = errors.Errorf("error running command '%v %v': %v", executable, strings.Join(args, " "), err)
```
Comment (neolit123) [inline cmd/clusterctl/clusterdeployer/bootstrap/minikube/minikube.go]: `Wrapf()` should be used in all places where there is an error wrapped, but also extra args.
https://github.com/pkg/errors/blob/master/errors.go#L198

File: cmd/clusterctl/clusterdeployer/bootstrap/minikube/minikube_test.go [hunk 2]
```diff
@@ -34,7 +35,7 @@ func TestCreate(t *testing.T) {
 		},
 		{
 			name:      "exec fail",
-			execError: fmt.Errorf("test error"),
+			execError: errors.Errorf("test error"),
```
Comment (neolit123) [inline cmd/clusterctl/clusterdeployer/bootstrap/minikube/minikube_test.go]: `errors.New()` should be used in all places where there are not arguments.

File: cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go [hunk 3]
```diff
@@ -110,7 +109,7 @@ func (c *client) removeKubeconfigFile() error {
 func (c *client) EnsureNamespace(namespaceName string) error {
 	clientset, err := clientcmd.NewCoreClientSetForDefaultSearchPath(c.kubeconfigFile, clientcmd.NewConfigOverrides())
 	if err != nil {
-		return fmt.Errorf("error creating core clientset: %v", err)
+		return errors.Errorf("error creating core clientset: %v", err)
```
Comment (neolit123) [inline cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go]: `errors.Wrap` should be used in all places where we are wrapping and there are no extra args.

**Comment by vincepri** [comment]
@neolit123 Thanks for clarifying! PR is updated 😊

File: pkg/controller/machineset/controller.go [hunk 4]
```diff
@@ -183,9 +184,9 @@ func (r *ReconcileMachineSet) Reconcile(request reconcile.Request) (reconcile.Re
 	updatedMS, err := updateMachineSetStatus(r.Client, machineSet, newStatus)
 	if err != nil {
 		if syncErr != nil {
-			return reconcile.Result{}, fmt.Errorf("failed to sync machines. %v. failed to update machine set status. %v", syncErr, err)
+			return reconcile.Result{}, errors.Wrapf(err, "failed to sync machines: %v. failed to update machine set status", syncErr)
```
Comment (neolit123) [inline pkg/controller/machineset/controller.go]: so this is a case that the Wrap functions do not solve - returning a combination of two errors.
but the real problem is that this function should be returning a list of errors, ideally, because it does have such an edge case.

i think this change is mostly OK (ish).

**Comment by vincepri** [comment]
@neolit123 ptal

**Comment by vincepri** [comment]
/assign @roberthbailey 
@neolit123 This should be good to go.

File: scripts/ci-is-vendor-in-sync.sh [hunk 5]
```diff
@@ -17,6 +17,7 @@
 set -o errexit
 set -o nounset
 set -o pipefail
+set -o verbose
```
Comment (neolit123) [inline scripts/ci-is-vendor-in-sync.sh]: this change seems out of scope of the PR?

Comment (neolit123) [reply scripts/ci-is-vendor-in-sync.sh]: the rest is LGTM.

Comment (vincepri) [reply scripts/ci-is-vendor-in-sync.sh]: You're correct, that was a leftover from testing

Comment (vincepri) [reply scripts/ci-is-vendor-in-sync.sh]: Fixed now
